lib/tokenizer.js

```diff
@@ -295,19 +436,115 @@ class ByteOrderedTokenizer extends Tokenizer {
   }
 
   /**
-   * @param {Buffer|Array} value
+   * @param {Buffer} value
    * @returns {Buffer}
    */
   hash(value) {
-    return value;
+    // strip any trailing zeros as tokens with trailing zeros are equivalent
+    // to those who don't have them.
+    let zeroIndex = value.length;
+    for(let i = value.length - 1; i > 0; i--) {
+      if(value[i] === 0) {
+        zeroIndex = i;
+      } else {
+        break;
+      }
+    }
+    return new token.ByteOrderedToken(value.slice(0, zeroIndex), this);
   }
 
   stringify(value) {
-    return value.toString('hex');
+    return value.toString('hex').toUpperCase();
   }
 
   parse(value) {
-    return utils.allocBufferFromString(value);
+    return this.hash(utils.allocBufferFromString(value, 'hex'));
```

Comment:
I think this might have been previously broken, the tokens returned from C* are hexstrings so should be parsed that way:

```
Datacenter: datacenter1
==========
Address    Rack        Status State   Load            Owns                Token                                       
                                                                          56713727820156410577229101238628035242      
127.0.0.1  rack1       Up     Normal  46.66 KB        66.67%              00                                          
127.0.0.3  rack1       Up     Normal  46.69 KB        66.67%              0113427455640312821154458202477256070484    
127.0.0.2  rack1       Up     Normal  46.69 KB        66.67%              56713727820156410577229101238628035242  
```